### PR TITLE
WIP: Add rest api

### PR DESCRIPTION
**Pull request by rake5k**

### CODE REVIEW

File: ansible/bootstrap.yml [hunk 1]
```diff
@@ -21,7 +21,7 @@
         packages:
           - mpd
           - mpc
-          - openjdk-8-jdk-headless
```
Comment (0xCA5A) [inline ansible/bootstrap.yml]: to be discussed -> ZERO compatibility due to non existing FPU
 * JAVA 8 lock
 * alternative JVM

File: ansible/deploy.yml [hunk 2]
```diff
@@ -20,6 +20,7 @@
       systemd:
         name: '{{ jbert_application_name }}'
         state: stopped
+      ignore_errors: yes
```
Comment (0xCA5A) [inline ansible/deploy.yml:23]: thx!

File: application.local.conf [hunk 3]
```diff
@@ -0,0 +1,14 @@
+backend {
+  halMockEnabled = true
+}
+
+frontend {
+  api {
+    playlists {
+      location = "/Users/christian/.mpd/playlists"
```
Comment (0xCA5A) [inline application.local.conf]: No host related paths

File: application/app/models/Playlist.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
+package models;
+
+import java.util.Objects;
+
+public class Playlist implements Comparable<Playlist> {
+
+    private PlaylistDto wrapped;
+
+    private Playlist(PlaylistDto wrapped) {
```
Comment (0xCA5A) [inline application/app/models/Playlist.java:9]: Remove Dto postfix in DTO naming? The postfix is defined in build.sbt

./build.sbt:18:    openapiJavaConfig := JavaConfig.default.copy(json = JavaJson.jackson).addMapping(CustomMapping.forName({ case s => s + "Dto" })),

Comment (0xCA5A) [reply application/app/models/Playlist.java:9]: It does not look so bad with the Dto postfix!

File: application/app/modules/ApplicationModule.java [hunk 5]
```diff
@@ -1,23 +1,46 @@
...
 
+import javax.inject.Named;
 import java.util.logging.Logger;
 
 
 public class ApplicationModule extends AbstractModule {
-    private static final Logger logger = LogHelper.getLogger(ApplicationModule.class.getName());
 
-    private final Config config = ConfigFactory.load();
+    private static final Logger logger = LogHelper.getLogger(ApplicationModule.class.getName());
```
Comment (0xCA5A) [inline application/app/modules/ApplicationModule.java]: Lets switch to SLF4J - ill kick out the LogHelper soon

File: application/app/modules/ApplicationModule.java [hunk 6]
```diff
@@ -1,23 +1,46 @@
...
+    private static final String PLAYLISTS = "playlists";
+    private static final String TRACKS = "tracks";
 
     @Override
     protected void configure() {
         printBuildInfo();
 
-        install(new BackendModule(config.getConfig("backend")));
+        final Config backendConfig = config.getConfig("backend");
+        final Config apiConfig = config.getConfig("frontend.api");
```
Comment (0xCA5A) [inline application/app/modules/ApplicationModule.java]: Is it possible to add the keys to reference.conf?
At least the structure...

File: application/app/modules/ApplicationModule.java [hunk 7]
```diff
@@ -1,23 +1,46 @@
...
+
+        install(new BackendModule(backendConfig));
+
+        bind(Config.class).annotatedWith(Names.named(PLAYLISTS)).toInstance(apiConfig.getConfig(PLAYLISTS));
+        bind(Config.class).annotatedWith(Names.named(TRACKS)).toInstance(apiConfig.getConfig(TRACKS));
+    }
+
+    @Provides
+    PlaylistService providePlaylistService(@Named(PLAYLISTS) Config config, TrackService trackService) {
```
Comment (0xCA5A) [inline application/app/modules/ApplicationModule.java]: Picky: Config naming

File: application/app/modules/ApplicationModule.java [hunk 8]
```diff
@@ -1,23 +1,46 @@
...
+        return new PlaylistService(config, trackService);
+    }
+
+    @Provides
+    TrackService provideTrackService(@Named(TRACKS) Config config) {
```
Comment (0xCA5A) [inline application/app/modules/ApplicationModule.java]: Picky: Config naming

File: runLocal.sh [hunk 9]
```diff
@@ -0,0 +1,3 @@
+#!/usr/bin/env bash
+
+sbt -Dconfig.file=application.local.conf ~jbert/run
```
Comment (0xCA5A) [inline runLocal.sh:3]: top!

File: application/app/models/Playlist.java (hunk 4 above)
Comment (rake5k) [reply application/app/models/Playlist.java:9]: This class is to extend the PlaylistDto with methods like a custom `equals()` and the `Comparable` interface. If you have a better idea how to make a `PlaylistDto` sortable, I'm open for proposals :-)

Comment (0xCA5A) [reply application/app/models/Playlist.java:9]: Got it. I was more in to remove the Dto postfix, but I would leave it as it is for now.

**Pull request closed without merging**